openqabot/types/aggregate.py

```diff
@@ -57,157 +57,158 @@ def get_buildnr(repohash: str, old_repohash: str, build: str) -> str:
+            query = settings["PUBLIC_CLOUD_TOOLS_IMAGE_QUERY"]
+            settings = apply_pc_tools_image(settings)
+            if not settings.get("PUBLIC_CLOUD_TOOLS_IMAGE_BASE", False):
+                log.error(
+                    f"Failed to query latest publiccloud tools image using {query}"
+                )
+                return {}
+
+        # parse Public-Cloud image REGEX if present
+        if "PUBLIC_CLOUD_IMAGE_REGEX" in settings:
+            settings = apply_publiccloud_regex(settings)
+            if not settings.get("PUBLIC_CLOUD_IMAGE_LOCATION", False):
+                log.error(
+                    f"No publiccloud image found for {settings['PUBLIC_CLOUD_IMAGE_REGEX']}"
+                )
+                return {}
+        # parse Public-Cloud pint query if present
+        if "PUBLIC_CLOUD_PINT_QUERY" in settings:
+            settings = apply_publiccloud_pint_image(settings)
+            if not settings.get("PUBLIC_CLOUD_IMAGE_ID", False):
+                log.error(
+                    f"No publiccloud image fetched from pint for for {settings['PUBLIC_CLOUD_PINT_QUERY']}"
+                )
+                return {}
+
+        full_post["openqa"].update(settings)
+        full_post["openqa"]["FLAVOR"] = self.flavor
+        full_post["openqa"]["ARCH"] = arch
+        full_post["openqa"]["_OBSOLETE"] = 1
+
+        for template, issues in test_incidents.items():
+            full_post["openqa"][template] = ",".join(str(x) for x in issues)
+        for template, issues in test_repos.items():
+            full_post["openqa"][template] = ",".join(issues)
+        for issues in test_incidents.values():
+            full_post["qem"]["incidents"] += issues
+
+        full_post["qem"]["incidents"] = [
+            str(inc) for inc in set(full_post["qem"]["incidents"])
+        ]
+
+        if not full_post["qem"]["incidents"]:
+            return {}
+
+        full_post["openqa"]["__DASHBOARD_INCIDENTS_URL"] = ",".join(
+            f"https://dashboard.qam.suse.de/incident/{inc}"
+            for inc in set(full_post["qem"]["incidents"])
+        )
+        full_post["openqa"]["__SMELT_INCIDENTS_URL"] = ",".join(
+            f"https://smelt.suse.de/incident/{inc}"
+            for inc in set(full_post["qem"]["incidents"])
+        )
+
+        full_post["qem"]["settings"] = full_post["openqa"]
+        full_post["qem"]["repohash"] = full_post["openqa"]["REPOHASH"]
+        full_post["qem"]["build"] = full_post["openqa"]["BUILD"]
+        full_post["qem"]["arch"] = full_post["openqa"]["ARCH"]
+        full_post["qem"]["product"] = self.product
+
+        return full_post
 
-        return ret
+    def __call__(
+        self,
+        incidents: List[Incident],
+        token: Dict[str, str],
+        ci_url: Optional[str],
+        ignore_onetime: bool = False,
+    ) -> List[Dict[str, Any]]:
+        return [
+            self.handle_arch(incidents, token, ci_url, ignore_onetime)
+            for arch in self.archs
```

Comment:
I think in the previous PR we saw that `arch` is not defined in `handle_arch` which resulted in a regression in production.
`arch` should be passed as an additional argument.

Reply:
I thought the regression was fixed in https://github.com/openSUSE/qem-bot/pull/89

Reply:
but, yes, the arch argument is still missing. I am more concerned if no CI checks fail and we don't see a report about missing coverage

Reply:
The regression that was fixed was that it should be called as a method, but after we deployed that we still saw that `arch` was undefined.
And then we reverted the whole PR...

Reply:
I am and was actually concerned about the fact why this compiled but the `arch` variable was undefined.
And I wasn't able to reproduce this locally with a simple function. In that case `arch` was defined.